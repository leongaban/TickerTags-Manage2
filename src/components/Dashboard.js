import React from 'react'
import { getCookie } from '../utils/session'
import { getMyTickers } from '../services/tickersFactory'
import TopSearchBar from './common/TopSearchBar'
import SidebarMenu from './sidebars/SidebarMenu'
import SidebarTasks from './sidebars/SidebarTasks'
import TickersTables from './tables/TickersTables'

const milliseconds = (new Date).getTime();

const userModel = {
	id: 133,
	username: 'leongaban',
	role: 'Admin'
};

const tickersFetchParams = {
	user: {},
    type: 'mine',
    tickers: [],
    service_start: 0,
    service_order: 'ticker',
    total: 0,
    date: milliseconds
};

export default function Dashboard(props) {
	let tickers = [];
	tickersFetchParams.user = userModel;

	getMyTickers(tickersFetchParams).then((myTickers) => {
		tickers = myTickers;
	});

	return (
		<div className="h100">
			<TopSearchBar />
			<section id="manage-dashboard">
				<SidebarMenu />
				<TickersTables tickers={ tickers }/>
				<SidebarTasks />
			</section>
		</div>
	)
}