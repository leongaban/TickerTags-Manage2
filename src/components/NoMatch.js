import React from 'react'

const NoMatch = () => {
	return (
		<div className="app-bg">
			<section id="auth-section">
				<header>
	            	<img src="static/imgs/logo.png"/>
	            	<h1>Oops!<br/> We could not find that...</h1>
	            	<h2>Try the links below:</h2>
	        	</header>
	        	<section className="missing-links">
	        		<ul>
	        			<li>
	        				<a href="/">Login</a>
	        			</li>
	        			<li>
	        				<a href="/forgot">Forgot Password</a>
	        			</li>
	        			<li>
	        				<a href="https://www.tickertags.com/help.html" target="_blank">TickerTags Help</a>
	        			</li>
	        			<li>
	        				<a href="https://www.tickertags.com" target="_blank">Marketing site</a>
	        			</li>
	        		</ul>
	        	</section>
	        	<footer> &copy; { new Date().getFullYear() } TickerTags</footer>
        	</section>
		</div>
	)
}

export default NoMatch