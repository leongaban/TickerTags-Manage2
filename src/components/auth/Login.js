import React from 'react'
import jwt from 'jsonwebtoken'
import { not } from 'ramda'
import { getLoginInputs, loginFormFilled, validEmail } from '../../utils/validation'
import * as session from '../../utils/session'
// import { setCurrentUser } from '../../actions/authActions'
import { error401, invalidLogin, notifyDev, missingLogin } from '../../consts/errors'
import { tagline } from '../../consts/sentences'
import * as api from '../../services/api'
import Notification from '../common/Notification'

class Login extends React.Component {
	constructor(props) {
		super(props)
		// console.log('props', props);

        this.state = {
            username: '',
            password: '',
            error: false,
            errorMsg: ''
        };

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
        this.closeMessage = this.closeMessage.bind(this);
	}

    handleChange(e) {
    	this.setState({ [e.target.name]: e.target.value });
    }

    handleSubmit(e) {
        e.preventDefault();
        const user = this.state.username;
        const pass = this.state.password;

        const gotoDashboard = (res) => {
			const user = res.data.user;
        	const ws_token_cookie = session.getCookie("_ws_token");
        	// console.log('ws_token_cookie', ws_token_cookie);
        	// session.addLocalStorage(ws_token_cookie);
        	// session.setAuthorizationToken(ws_token_cookie);
        	console.log('user', user);
        	// this.props.setUser(user);
        	// dispatch(setCurrentUser(user));

        	// console.log(' jwt', jwt);
        	// console.log('  jwt.decode(ws_token_cookie)', jwt.decode(ws_token_cookie, { complete: true }));
        	this.props.history.push('/dashboard');
        };

        const postLogin = (user, pass) => {
			api.userLogin(user, pass)
				.then((res) => {
					if (res.toString() === error401) {
						return this.setState({
							error: true,
							errorMsg: invalidLogin
						});
					}

					if (res.status && res.status === 200) {
						this.setState({
							error: false,
							errorMsg: ''
						});
						gotoDashboard(res);
					}
					else {
						return this.setState({
							error: true,
							errorMsg: notifyDev
						});
					}
				});
		};

        loginFormFilled(user, pass)
        	? postLogin(user, pass)
        	: this.setState({
        		error: true,
        		errorMsg: missingLogin
        	});
    }

    closeMessage() {
    	this.setState({ error: false, errorMsg: '' });
    }

	render() {
		const usernameError = this.state.username.error;
		const error = this.state.error;
		const errorMsg = this.state.errorMsg;

		return (
			<div className="app-bg">
				{ error &&
					<Notification message={ errorMsg } closeMsg={ this.closeMessage }/>
				}
                <section id="auth-section">
                	<header>
	                	<img src="static/imgs/logo.png"/>
	                	<h1>{ tagline }</h1>
                	</header>
                	<div className="login-form">
	                	<form onSubmit={ this.handleSubmit } className="form-login">
	                        <ul>
	                            <li className="rel">
	                            	<div className="icon-user-circle-o"></div>
	                                <input type="text"
	                                       id="input-auth-username"
	                                       name="username"
	                                       placeholder="username"
	                                       onChange={ this.handleChange } />
	                            </li>
	                            <li className="rel">
	                            	<div className="icon-lock"></div>
	                                <input type="password"
	                                       id="input-auth-password"
	                                       name="password"
	                                       placeholder="password"
	                                       onChange={ this.handleChange } />
	                            </li>
	                            <li>
	                                <button className="btn-green" type="submit">Login</button>
	                            </li>
	                            <li>
	                                <a href="/forgot" className="small-auth-link">
	                                    <p>Forgot Password</p>
	                                </a>
	                                <footer> &copy; { new Date().getFullYear() } TickerTags</footer>
	                            </li>
	                        </ul>
	                    </form>
					</div>
                </section>
			</div>
		)
	}
}

export default Login