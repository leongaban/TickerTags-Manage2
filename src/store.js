import React from "react"
import { applyMiddleware, combineReducers, compose, createStore} from "redux"
import thunk from "redux-thunk"
import { userReducer } from "./reducers/UserReducer"
import { authReducer } from "./reducers/AuthReducer"

export const createCommonStore = (trackStore=false) => {
    const reducers = combineReducers({
        user: userReducer,
        user: authReducer
    });

    //noinspection JSUnresolvedVariable
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

    // const store = createStore(reducers,
    //     compose(
    //         applyMiddleware(thunk),
    //         // window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
    //     )
    // );

    const enhancer = composeEnhancers(
        applyMiddleware(thunk)
        // applyMiddleware(...thunk),
    );

    if (trackStore) {
        store.subscribe((() => {
            console.log("  store changed", store.getState());
        }));
    }

    const store = createStore(reducers, enhancer);

    return store;
};