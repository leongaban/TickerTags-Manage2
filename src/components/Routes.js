import React from 'react'
import { Route, Switch } from 'react-router-dom'
import LoginContainer from './auth/Login'
import Dashboard from './Dashboard'
import NoMatch from './NoMatch'

const Routes = () => {
    return (
        <Switch>
        	<Route exact={ true } path="/" component={ LoginContainer }/>
        	<Route path="/dashboard" component={ Dashboard }/>
        	<Route component={ NoMatch } />
        </Switch>
    );
}

export default Routes