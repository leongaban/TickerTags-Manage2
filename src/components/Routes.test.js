import React from 'react'
import { Provider } from "react-redux"
import { mount, shallow } from 'enzyme'
import { MemoryRouter } from 'react-router' // https://github.com/ReactTraining/react-router/blob/master/packages/react-router/docs/api/MemoryRouter.md
import { createCommonStore } from "../store";
import toJson from 'enzyme-to-json'
import Routes from './Routes'
import Login from './auth/Login'
import Dashboard from './Dashboard'
import NoMatch from './NoMatch'

const store = createCommonStore();

const routesMount = (path) => {
	return mount(
		<Provider store={store}>
			<MemoryRouter initialEntries={[ path ]} initialIndex={0}>
				<Routes />
			</MemoryRouter>
		</Provider>
	);
};

describe('<Routes /> component', () => {
	it('should save a snapshot', () => {
		const routesComponent = shallow(<Routes />);
		const tree = toJson(routesComponent);
		expect(tree).toMatchSnapshot();
	});

	it('should render Login component when visiting /', () => {
		const routesComponent = routesMount('/');
		expect(routesComponent.find(Login).length).toBe(1);
	});

	it('should render Dashboard component when visiting /dashboard', () => {
		const routesComponent = routesMount('/dashboard');
		expect(routesComponent.find(Dashboard).length).toBe(1);
	});

	it('should render the NoMatch component when visiting invalid path', () => {
		const routesComponent = routesMount('/404');
		expect(routesComponent.find(NoMatch).length).toBe(1);
	});
});