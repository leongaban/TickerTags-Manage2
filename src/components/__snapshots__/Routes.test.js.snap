// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Routes /> component should save a snapshot 1`] = `
<Switch>
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    component={[Function]}
    path="/dashboard"
  />
  <Route
    component={[Function]}
  />
</Switch>
`;
